src/Android/Effects/NoEmojiKeyboardEffect.cs

```diff
@@ -0,0 +1,24 @@
+﻿using Bit.Droid.Effects;
+using Xamarin.Forms;
+using Xamarin.Forms.Platform.Android;
+
+[assembly: ExportEffect(typeof(NoEmojiKeyboardEffect), "NoEmojiKeyboardEffect")]
+namespace Bit.Droid.Effects
+{
+    public class NoEmojiKeyboardEffect : PlatformEffect
+    {
+        protected override void OnAttached()
+        {
+            if (Element is Entry && Control is FormsEditText editText)
+            {
+                editText.ImeOptions = Android.Views.InputMethods.ImeAction.Done;
+                editText.InputType = Android.Text.InputTypes.ClassText | Android.Text.InputTypes.TextVariationVisiblePassword | Android.Text.InputTypes.TextFlagMultiLine;
+            }
```

Comment:
Why do you need to check whether if the `Element` is an `Entry`? I'd just check the `Control` so that the effect can be applied to anything that has a control that is `FormsEditText`. As a matter of fact, I'd just change it to the base class `EditText`.
Aside from that, why does this effect changes the `ImeOptions`? IMO, this is like adding a "hidden" change in the effect because unless you read this code you don't know that it also changes it. Furthermore, you can control the `ImeOptions` [directly on the xaml](https://docs.microsoft.com/en-us/xamarin/xamarin-forms/platform/android/entry-ime-options) so it's not needed here.

Reply:
Removed the ImeOptions as they are not necessary for this